### PR TITLE
Add warning about Katacoda shutdown

### LINKED ISSUE #34004: Warn about Katacode shutdown
**This is a Feature Request**

**What would you like to be added**
For pages that use Katacoda, add a warning about its removal.

**Why is this needed**
Prompted by https://github.com/kubernetes/website/issues/33918 and https://www.oreilly.com/online-learning/leveraging-katacoda-technology.html

**Comments**
/priority critical-urgent
/language en
/triage accepted

### PR DESCRIPTION
**Pull request by sftim**
Fixes #34004

Add a warning about the shutdown of Katacoda (see #33936 and #33918).

---

[Example](https://deploy-preview-34005--kubernetes-io-main-staging.netlify.app/docs/tutorials/hello-minikube/)

### CODE REVIEW

**Comment by sftim** [comment]
@tengqm - you're this week's PR wrangler and @bradtopol - you're up from tomorrow - could I ask you to move this one forward if you get time?

File: layouts/partials/katacoda-removal.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<section id="katacoda-removal-warning">
+  <div class="content warning pageinfo pageinfo-warning">
+    <p>
+      {{ T "katacoda_removal_warning" | markdownify }}
+    </p>
+  </div>
+</section>
```
Comment (Sea-n) [inline layouts/partials/katacoda-removal.html:7]: Maybe we can add a newline in the end of this file.

File: layouts/docs/baseof.html [hunk 2]
```diff
@@ -24,6 +24,11 @@
             {{ if not .Site.Params.ui.breadcrumb_disable }}{{ partial "breadcrumb.html" . }}{{ end }}
             {{ block "deprecated" . }}
               {{ partial "deprecation-warning.html" . }}
+            {{ end }}
+            {{ if .HasShortcode "kat-button" }}
```
Comment (tengqm) [inline layouts/docs/baseof.html:28]: Not sure how to make the condition more generic. For example, page 'https://deploy-preview-34005--kubernetes-io-main-staging.netlify.app/docs/tutorials/kubernetes-basics/explore/explore-interactive/' doesn't have such a warning.

A more straightforward approach would be add this warning to those pages directly, and then remove the whole page completely when the day comes.

Comment (sftim) [reply layouts/docs/baseof.html]: I've added a new shortcode to those pages. We can now detect all Katacoda use for the English localization, and updating other localizations is only a search-and-replace away.

**Comment by mtardy** [comment]
/lgtm
Aside I just noticed that the CSS is messed up on these Katacoda tutorial pages like this one https://kubernetes.io/docs/tutorials/kubernetes-basics/explore/explore-interactive/.

It's not related to this PR, but it's obvious with the new warning box:
https://deploy-preview-34005--kubernetes-io-main-staging.netlify.app/docs/tutorials/hello-minikube/
https://deploy-preview-34005--kubernetes-io-main-staging.netlify.app/docs/tutorials/kubernetes-basics/explore/explore-interactive/

**Pull request merged**